### PR TITLE
Fix: creating a Jira issue

### PR DESCRIPTION
**Pull request by BartoszJarocki**
The endpoint we use to convert markdown to adf doesn't work anymore. Calling `manager.convertMarkdownToADF(markdown)` returns 
```
Error: No message available
      at AtlassianServerManager.post (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:78329:16)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Promise.all (index 1)
      at async createJiraTask (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:12503:39)
      at async Object.resolve [as MutationcreateJiraTaskIntegrationResolver] (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:8454:17),
```
 We don't check errors in the response and as a result, we're trying to create a task with the following payload
```
{
  summary: "I'm a task",
  description: Error: No message available
      at AtlassianServerManager.post (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:78329:16)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Promise.all (index 1)
      at async createJiraTask (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:12503:39)
      at async Object.resolve [as MutationcreateJiraTaskIntegrationResolver] (/Users/bartoszjarocki/Work/parabol/web-parabol/dev/gqlExecutor.js:8454:17),
  assignee: { id: '60ede678a211b100694a405b' },
  issuetype: { id: '10002' }
}
```
And the Jira API complains. 

This PR is a little proof of concept that this is the root cause of the reported customer issue 
Screenshot 2021-09-23 at 07 46 07.

I replaced `manager.convertMarkdownToADF(markdown)` with a package https://github.com/b-yond-infinite-network/md-to-adf and it works. The only problem is, the package isn't actively supported so I'm not sure if it's safe to use. I haven't found anything better so any ideas are more than welcome!

### CODE REVIEW

File: packages/server/graphql/mutations/helpers/createJiraTask.ts [hunk 1]
```diff
@@ -28,7 +26,7 @@ const createJiraTask = async (
   const bestType = issuetypes.find((type) => type.name === 'Task') || issuetypes[0]
   const payload = {
     summary,
-    description,
+    description: convertMdToAdf(markdown),
```
Comment (Dschoordsch) [inline packages/server/graphql/mutations/helpers/createJiraTask.ts:29]: If we skip the conversion to markdown, but instead work with [`contentState.getBlockMap()`](https://draftjs.org/docs/api-reference-content-state#getblockmap) directly, it should be quite simple to convert these few [`ContentBlock` types](https://draftjs.org/docs/api-reference-content-block) to [ADF](https://developer.atlassian.com/cloud/jira/platform/apis/document/structure/) directly.

**Review by Dschoordsch — COMMENTED** [review]
The b-yond people are using the library themselves [ga-sync-issues-to-jira](https://github.com/b-yond-infinite-network/ga-sync-issues-to-jira), so probably support it, but it also depends on this [adf-builder-javascript](https://github.com/LucasRitter/adf-builder-javascript) which is even older. I propose doing the conversion manually.

**Comment by mattkrick** [comment]
waking up & finding a solution to this is the best feeling in the world!
testing now...

**Comment by BartoszJarocki** [comment]
@mattkrick I think we still need to add support for nested lists. @Dschoordsch did a great job implementing our own conversion, and this seems to be the last piece left. 🙏

File: packages/server/graphql/mutations/helpers/convertContentStateToADF.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            ...(data.title ? {title: data.title} : {})
+          }
+        }
+        return contentPieces.map((piece) => ({
+          ...piece,
+          marks: [mark, ...(piece.marks ?? [])]
+        }))
+      }
+      /*
+      TODO: In order to show images or other embedded types, we need to upload those separately to Jira and then refer to them by id
```
Comment (mattkrick) [inline packages/server/graphql/mutations/helpers/convertContentStateToADF.ts:101]: AOK to skip, our draftjs implementation doesn't support images, so we couldn't push em even if we wanted to

**Review by mattkrick — APPROVED** [review]

**Comment by mattkrick** [comment]
Test result here: https://parabol-2.atlassian.net/browse/DEV3-129

Conversation on atlassian here: https://community.atlassian.com/t5/Jira-questions/convert-jira-ADF-formatted-description-in-HTML-format/qaq-p/1750152

**Pull request merged**